### ISSUE TITLE
Insert data interpretation layer

### ISSUE THREAD

**Issue by jefffhaynes** [jefffhaynes/BinarySerializer#77]
At some point in the distant future, remove weird work-around for deferred evaluation of endianness and add an interpretation layer that decouples reading and writing from value representation

**Comment by jefffhaynes** [comment]
No, the idea is simply to decouple the interpretation of the data (e.g converting four bytes into an int) from the reading of the data. This is especially relevant when endianness comes into play and there are some hacky things happening in the current design because the decoupling isn’t that clean right now.

—
If you want to build a ship, don't drum up people to collect wood and don't assign them tasks and work, but rather teach them to long for the endless immensity of the sea.

Antoine de Saint-Exupery
________________________________

**Comment by Waelwindows** [comment]
Ah, so will this help with the FieldScale bug you have with a different endianness?

**Comment by jefffhaynes** [comment]
Maybe. What bug is that? 🙂

**Comment by Waelwindows** [comment]
If you have a field with the FieldScale attribute in Little Endian it deserializes correctly, however, if you try the same but in BigEndian, you get garbled data.

**Comment by jefffhaynes** [comment]
Then yes, that’s probably related. Would you mind entering that as a separate bug if it isn’t already?

**Comment by jefffhaynes** [comment]
Thanks!

________________________________
